### DISCUSSION TITLE
Error in ‘reducedDim’ for UMAP

### DISCUSSION THREAD

**Discussion by karlaarz** [campbio/celda#350] [Q&A] [answered]
Hello, 

I'm trying to run the latest version of decontX using the dgCMAtrix from seurat. The analysis runs smoothly and I get the decontx list output. However I can't do the UMAP analysis and the rest of the plots and analyses showed in the vignette. 

`seur.dtc <- decontX(seur.dtc) `
`plotDecontXContamination(seur.dtc)`
`umap <- reducedDim(seur.dtc, "decontX_UMAP")`

`Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function ‘reducedDim’ for signature ‘"list", "character"’`

Am I missing something?

**Comment by joshua-d-campbell** [comment, marked as answer]
Hi @karlaarz, thanks for trying out our tool. If the input is just a `dgCMatrix`, then the output will be just be a list of results. The `reducedDim` function expects a `SingleCellExperiment` (SCE) object as input. If you put your counts into a SCE object first, then your code should work:

```
library(SingleCellExperiment)
sce <- SingleCellExperiment(list(counts = seur.dtc))
sce <- decontX(sce)
umap <- reducedDim(seur.dtc, "decontX_UMAP")
```

The rest of the plots should work the same as outlined in our [vignette](https://bioconductor.org/packages/release/bioc/vignettes/celda/inst/doc/decontX.pdf). If you don't want to use the SCE, then you can get the UMAP and clusters from the result list:

```
res <- decontX(seur.dtc)
umap <- res$estimates$all_cells$UMAP

# Plot clusters
plotDimReduceCluster(x = res$z, dim1 = umap[, 1], dim2 = umap[, 2])

# Plot contamination
plotDecontXContamination(res)
```

Note that you will want to save the decontX result into a variable with another name so you don't overwrite your original matrix. I'm going to move this to the Discussions section instead of an issue in case other people have the same question. Just let me know if you have other issues or questions.

**Reply by karlaarz** [reply]
Hi @joshua-d-campbell thanks for your reply. Indeed, when running as SCE, the rest of the vignette worked smoothly.
